### PR TITLE
add quasi88, a NEC PC-8801 emulator

### PR DESCRIPTION
**Pull request by nakatamaho**
add quasi88, a NEC PC-8801 emulator

### CODE REVIEW

File: platforms.cfg [hunk 1]
```diff
@@ -164,6 +164,9 @@ psx_fullname="PlayStation"
 ps2_exts=".iso .img .bin .mdf .z .z2 .bz2 .dump .cso .ima .gz"
 ps2_fullname="PlayStation 2"
 
+quasi88_exts=".d88 .D88 .cmt .CMT .t88 .T88"
+quasi_fullname="PC-8801"
```
Comment (joolswills) [inline platforms.cfg]: should be quasi88_fullname

File: scriptmodules/emulators/quasi88.sh [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+#
+# See the LICENSE.md file at the top-level directory of this distribution and
+# at https://raw.githubusercontent.com/RetroPie/RetroPie-Setup/master/LICENSE.md
+#
+
+rp_module_id="quasi88"
+rp_module_desc="NEC PC-8801 emulator"
+rp_module_help="ROM Extensions: .d88 .D88 .cmt .CMT .t88 .T88."
+rp_module_section="exp"
+version="0.6.4"
```
Comment (joolswills) [inline scriptmodules/emulators/quasi88.sh]: this should be removed

File: scriptmodules/emulators/quasi88.sh [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+function depends_quasi88() {
+    local depends=(libsdl1.2-dev)
+    getDepends "${depends[@]}"
```
Comment (joolswills) [inline scriptmodules/emulators/quasi88.sh]: you can just do `getDepends libsdl1.2-dev` in this case, but it's not a problem.

File: scriptmodules/emulators/quasi88.sh [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+}
+
+function sources_quasi88() {
+    wget -q -O- "http://www.eonet.ne.jp/~showtime/quasi88/release/quasi88-${version}.tgz" | tar -xvz --strip-components=1
+    cat > retropie_fixes.diff <<\_EOF_
```
Comment (joolswills) [inline scriptmodules/emulators/quasi88.sh]: for inline patches use applyPatch

use unified diff format.

File: scriptmodules/emulators/quasi88.sh [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    make
+}
+
+function install_quasi88() {
+    make install
+}
+
+function configure_quasi88() {
+    mkRomDir "quasi88"
+    mkUserDir "$md_conf_root/quasi88"
```
Comment (joolswills) [inline scriptmodules/emulators/quasi88.sh]: is this folder needed ? if the emulator has a config folder in the home directly it needs to be symlinked with "moveConfigDir"

File: platforms.cfg (hunk 1 above)
Comment (joolswills) [inline platforms.cfg]: also lowercase extensions only here.

File: scriptmodules/emulators/quasi88.sh (hunk 2 above)
Comment (joolswills) [inline scriptmodules/emulators/quasi88.sh]: also needs lowercase extensions.

File: scriptmodules/emulators/quasi88.sh (hunk 4 above)
Comment (joolswills) [inline scriptmodules/emulators/quasi88.sh]: local version can be included here or just hardcode it.

File: scriptmodules/emulators/quasi88/01_Makefile.diff [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
++SDL_VERSION	= 1
+ # GTK_VERSION	= 1
+ 
+ 
+@@ -32,7 +32,7 @@
+ #	�ǥ��쥯�ȥ꤬���Ѥ���ޤ���
+ #	 ~/ �ϡ�QUASI88�ε�ư���˥ۡ���ǥ��쥯�ȥ��$HOME�ˤ�Ÿ������ޤ���
+ 
+-ROMDIR	= ~/quasi88/rom/
++ROMDIR	= /home/pi/RetroPie/BIOS/quasi88
```
Comment (joolswills) [inline scriptmodules/emulators/quasi88/01_Makefile.diff]: This shouldn't be hardcoded as RetroPie can be used for any user.

You can put the patch in-line and use romdir/BIOS/quasi88, or put some string here and sed the patch before applying.

File: scriptmodules/emulators/quasi88/01_Makefile.diff [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+ 
+ 
+ # DISK�ѥǥ��쥯�ȥ�����ꤷ�ޤ�
+@@ -42,7 +42,7 @@
+ #	�ǥ��쥯�ȥ꤬���Ѥ���ޤ���
+ #	 ~/ �ϡ�QUASI88�ε�ư���˥ۡ���ǥ��쥯�ȥ��$HOME�ˤ�Ÿ������ޤ���
+ 
+-DISKDIR	= ~/quasi88/disk/
++DISKDIR	= /home/pi/RetroPie/roms/pc88
+
```
Comment (joolswills) [inline scriptmodules/emulators/quasi88/01_Makefile.diff]: as above

**Comment by joolswills** [comment]
actually, there is no need to patch the Makefile - it should be possible to override these makefile variables when calling MAKE - which would simplify it a lot.

**Comment by joolswills** [comment]
eg `make VARIABLE=` (you can combine this with a smaller patch if needed - but it can be used for the PATHS etc)

**Comment by joolswills** [comment]
eg something like (just an example - untested)

```
make X11_VERSION=0 SDL_VERSION=1 ROMDIR="$romdir/BIOS/quasi88" DISKDIR="$romdir/pc88" TAPEDIR="$romdir/pc88" USE_OLD_MAME_SOUND=1 BINDIR="$md_inst"
```

**Comment by nakatamaho** [comment]
* fixed issues mentiond avobe. avoid hardcoded paths.
* somehow make VARIABLE= didn't work. only ROM_DIR, DISK_DIR and TAPE_DIR work.
following is quote from Makefile.
CFLAGS += -DROM_DIR='"$(ROMDIR)"' -DDISK_DIR='"$(DISKDIR)"' \
          -DTAPE_DIR='"$(TAPEDIR)"' $(USEINLINE) -DCLIB_DECL=

**Comment by joolswills** [comment]
Which variable didnt work ? BINDIR should work ok for example.

**Comment by nakatamaho** [comment]
thanks for your comment. now I understand how to use command line options.
could you please review again?
Best regards.

**Comment by joolswills** [comment]
I added this with some changes:

* More help info
* Allow building for X11 but exclude Mali
* Rename patch and include fix for psg.cpp on later gcc
* Change platforms.cfg and configure_ to use pc88 (system name rather than emulator name)
* Change config location and launch commandline

**Comment by joolswills** [comment]
Cheers.

**Comment by nakatamaho** [comment]
Thank you very very much! you're kind person!

**Pull request closed without merging**